charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -327,6 +328,11 @@ def write(self):
             nrpe_monitors[nrpecheck.shortname] = {
                 "command": nrpecheck.command,
             }
+            # If we were passed max_check_attempts, add that to the relation data
+            try:
+                nrpe_monitors[nrpecheck.shortname]['max_check_attempts'] = nrpecheck.max_check_attempts
```

Comment:
If I'm reading this code right, then the value of 'max_check_attempts' in the nrpe_monitors will be explicitly set to None, and the only test confirms that adding configuration for this new option works. I'd like to see a test with this not being set along with being set to confirm that it works, as expected, in both situations.

Reply:
Thanks for this - it might be that copying the entire test is overkill, but the more I thought about it the more I thought it was worthwhile.  I've also updated to add in modification of the template which is exported to /var for remote Nagios hosts to collect, though I don't know if anyone still uses that until we remove it entirely we should maintain it as current.